main.tf

```
@@ -1,13 +1,254 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
 
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+locals {
+  enabled = module.this.enabled
+
+  security_group_enabled = local.enabled && var.create_security_group
+  s3_bucket_enabled      = local.enabled && var.create_s3_bucket
+  iam_role_enabled       = local.enabled && var.create_iam_role
+  account_id             = data.aws_caller_identity.current.account_id
+  region                 = data.aws_region.current.name
+}
+
+module "s3_label" {
+  source  = "cloudposse/label/null"
+  version = "0.25.0"
+  enabled = local.s3_bucket_enabled
+
+  attributes = ["s3"]
+
+  context = module.this.context
+}
+
+module "sg_label" {
+  source  = "cloudposse/label/null"
+  version = "0.25.0"
+  enabled = local.security_group_enabled
+
+  attributes = ["sg"]
+
+  context = module.this.context
+}
+
+module "iam_label" {
+  source  = "cloudposse/label/null"
+  version = "0.25.0"
+  enabled = local.iam_role_enabled
+
+  attributes = ["iam"]
+
+  context = module.this.context
+}
+
+data "aws_iam_policy_document" "iam_policy_document" {
+  statement {
+    sid       = ""
+    actions   = ["airflow:PublishMetrics"]
+    effect    = "Allow"
+    resources = ["arn:aws:airflow:${local.region}:${local.account_id}:environment/${module.this.id}"]
```

```suggestion
    resources = ["arn:${local.aws_partition}:airflow:${local.region}:${local.account_id}:environment/${module.this.id}"]
```

let's support diff partitions (Gov, China, etc.)
use the partition data source to find the current partition and populate the local var `aws_partition`

---

Updated